client/blocks/login/index.jsx

```diff
@@ -3,169 +3,54 @@
-				log: this.state.usernameOrEmail,
-				pwd: this.state.password,
-				redirect_to: this.props.redirectLocation || window.location.origin,
-				rememberme: this.state.rememberme ? 1 : 0,
-			} );
-		} ).catch( errorMessage => {
-			this.props.recordTracksEvent( 'calypso_login_block_login_failure', {
-				error_message: errorMessage
-			} );
-
-			this.setState( {
-				submitting: false,
-				errorMessage
+				log: usernameOrEmail,
+				pwd: password,
+				redirect_to: redirectLocation || window.location.origin,
+				rememberme: rememberMe ? 1 : 0,
 			} );
-		} );
+		}
 	};
 
-	renderNotices() {
-		if ( this.state.errorMessage ) {
-			return (
-				<Notice status="is-error"
-					text={ this.state.errorMessage }
-					onDismissClick={ this.dismissNotice } />
-			);
-		}
-	}
+	handleValid2FACode = ( { twoStepCode, rememberMe } ) => {
+		twoStepCode; rememberMe;
+	};
 
 	render() {
-		const isDisabled = {};
-		if ( this.state.submitting ) {
-			isDisabled.disabled = true;
-		}
-
-		return (
-			<div className="login">
-
-				{ this.renderNotices() }
-
-				<div className="login__form-header">
-					<div className="login__form-header-title">
-						{ this.props.title }
-					</div>
-				</div>
-
-				<form onSubmit={ this.onSubmitForm }>
-					<Card className="login__form">
-						<div className="login__form-userdata">
-							<label htmlFor="usernameOrEmail" className="login__form-userdata-username">
-								{ this.props.translate( 'Username or Email Address' ) }
-							</label>
-							<FormTextInput
-								className="login__form-userdata-username-input"
-								onChange={ this.onChangeField }
-								id="usernameOrEmail"
-								name="usernameOrEmail"
-								value={ this.state.usernameOrEmail }
-								{ ...isDisabled } />
-							<label htmlFor="password" className="login__form-userdata-username">
-								{ this.props.translate( 'Password' ) }
-							</label>
-							<FormPasswordInput
-								className="login__form-userdata-username-password"
-								onChange={ this.onChangeField }
-								id="password"
-								name="password"
-								value={ this.state.password }
-								{ ...isDisabled } />
-						</div>
-						<div className="login__form-remember-me">
-							<label>
-								<FormCheckbox
-									name="rememberme"
-									checked={ this.state.rememberme }
-									onChange={ this.onChangeRememberMe }
-									{ ...isDisabled } />
-								<span>{ this.props.translate( 'Stay logged in' ) }</span>
-							</label>
-						</div>
-						<div className="login__form-action">
-							<FormsButton primary { ...isDisabled }>
-								{ this.props.translate( 'Log in' ) }
-							</FormsButton>
-						</div>
-					</Card>
-				</form>
-			</div>
-		);
+		const {
+			title,
+			twoFactorEnabled,
+		} = this.props;
+
+		return twoFactorEnabled
+			? ( <div
+					onSuccess={ this.handleValid2FACode } /> )
+			: ( <LoginForm
+					title={ title }
+					onSuccess={ this.handleValidUsernamePassword }
+				/> );
```

Comment:
This block of code seems to have exploded. What about returning to the line after each parenthesis as we usually do?

```javascript
return twoFactorEnabled ?
  (
    <div
      onSuccess={ this.handleValid2FACode } />
  )
  : (
    <LoginForm
      title={ title }
      onSuccess={ this.handleValidUsernamePassword } />
  );
```